crates/subspace-core-primitives/src/pieces.rs

```diff
@@ -298,12 +298,10 @@ impl AsRef<[u8]> for PieceIndexHash {
     }
 }
 
-impl PieceIndexHash {
-    // TODO: Remove and replace uses with `index.hash()`
-    /// Constructs `PieceIndexHash` from `PieceIndex`
+impl From<PieceIndex> for PieceIndexHash {
```

Comment:
Please don't do that. Such conversions should not be done with `From`. Also for some reason you have removed TODO that was aiming to replace this verbose method with a much more compact method call. You can already do `piece_index.hash()` instead of `PieceIndexHash::from(piece_index)` that you have implemented here. This `impl` should simply be removed, not updated.